howto/application/extend.md

```diff
@@ -1,38 +1,45 @@
-You can extend an application by adding an [addon](https://discourse.ubuntu.com/t/addons/25293) or [hooks](https://discourse.ubuntu.com/t/hooks/28555).
+You can extend an application either by adding [hooks](#application-hooks) directly to the application or by creating an [addon](#addon) that includes one or more hooks and adding it to the application.
 
-If your extension contains common functionality that you want to share among multiple applications, you should create an addon. You can then add the addon to all applications that should use the functionality.
+- If you want to extend one application only, you should use application hooks. They are easy and quick to set up and do not require creating an addon.
+- If your extension contains common functionality that you want to share among multiple applications, you should create an addon that includes one or more hooks. You can then add the addon to all applications that should use the functionality.
 
-If you want to extend one application only, use application hooks instead. They are easier to create and do not require creating an addon.
+For both options, you must create one or more hooks first. The options differ in how you add these hooks to your application.
 
-## Extend an application with addons
+<a name="create-hook"></a>
+## Create a hook
 
-See [Create an addon](https://discourse.ubuntu.com/t/creating-an-addon/25284) for detailed instructions.
+A hook is a script file that runs a series of commands at a specific time in the application life cycle. See [Hooks](https://discourse.ubuntu.com/t/hooks/28555) for detailed information.
 
-## Extend an application with hooks
+The general steps for creating a hook are as follows:
 
-The following example shows how to use application hooks to change the Android system locale.
+1. Create or download any files or applications that the hook needs. For example, this could be configuration files or Android applications that you want to run.
+1. In a `hooks/` sub-directory, create a script file with the commands that you want to run. As file name, use the name of the hook.
 
-1. In a new `app` directory, create a `manifest.yaml` file with the following content:
+   [note type="information" status="Tip"]
+   Supported hooks are `pre-start`, `post-start` and `post-stop`. See [Hooks](https://discourse.ubuntu.com/t/hooks/28555).
+   [/note]
```

Comment:
It would be better to add another note to highlight the usage of `CONTAINER_TYPE` environment variable that people tend to forget about it when creating a hook, which led to a bad situation where that hooks that are supposed to be only executed in a base container got executed in a regular container too.